### ISSUE TITLE
High CPU usage when just started, no containers running.

### ISSUE THREAD

**Issue by armori** [docker/for-mac#4392]
- [ x ] I have tried with the latest version of my channel (Stable or Edge)
  - [ x ] I have uploaded Diagnostics
  - Diagnostics ID: BCBAA646-C246-4F57-AB7D-2B54F05CC21D/20200325071455

### Expected behavior
Low CPU usage

### Actual behavior
CPU usage stuck arount 120-130% (i5 5257U)

### Information

  - macOS Version: 10.15.3

### Diagnostic logs

```
Docker for Mac: version 2.2.0.4

```

**Comment by cculianu** [comment]
Happens to me too. I get about 20% CPU usage on my i9 macbook pro.  It is just started, no containers running.   The `com.docker.hyperkit` process just eats CPU every few seconds, spiking it up to like 20%.  Then it goes to sleep and a few seconds later, the cycle repeats.

It causes the machine to heat up, fans to whirr, and for it to consume power needlessly.

**Comment by dnguyenv** [comment]
+1. Wondering if this is fixed in the recently released 2.2.0.5?

**Comment by engstrom** [comment]
> +1. Wondering if this is fixed in the recently released 2.2.0.5?

2.2.0.5 did not resolve the issue on my MacBook Pro (15-inch, 2017) running Catalina (10.15.3).

**Comment by cculianu** [comment]
Nope, not resolved for me either.  Docker just started 5 minutes ago.  No containers runnings.  It .. just sits there eating CPU for no reason.

# CPU Use

<img width="726" alt="Screen Shot 2020-04-03 at 10 17 22 AM" src="https://user-images.githubusercontent.com/266627/78334553-a0b44600-7594-11ea-9676-99640f2f8789.png">

# Energy Impact

<img width="727" alt="Screen Shot 2020-04-03 at 10 20 07 AM" src="https://user-images.githubusercontent.com/266627/78334615-bc1f5100-7594-11ea-8058-90b0fbbee6d5.png">

---

Mind you this is on a very fast i9 16 core system.  7% CPU use on one of these cores is like 25% on your typical i5 or i7.

I'm a C++ programmer.  I go way back to the early 2000's.  C or C++ programs that wake up randomly and churn the CPU that hard are programs I would consider broken and needing to be rewritten.

Go programs should be no different.

Docker likely wakes up and does some crazy mundane tasks every few seconds.  Or maybe it's jut the Go gc?

The devs should see about tuning the Go GC to not do that.

**Comment by zendraxl** [comment]
Same issue.

**Comment by nebuk89** [comment]
hey all, 

we will try and roll this into our upcoming CPU investigation but as highlighted on #12 this is likely a separate issue from the symptoms. 

@cculianu 

The com.docker.hyperkit process is a C program. If the disk is a qcow (if the system doesn't have APFS on an SSD) then there is also ocaml. 
However the ocaml heap is probably not that big. 
There's probably a command to introspect it's size (something like 'pmap -x pid' on Linux) to check this....  I would need to get a bit more help from the team to dig into this further to see if giving y'all a set of instructions to do self diagnostics on this is going to help you or something we should wait for us being a bit more ready to engage with :)

**Comment by cculianu** [comment]
Ah ok, C. Cool. Yeah I'm ok to try whatever you want.  Thanks for looking into it!

**Comment by UgurGumushan** [comment]
I have the same problem. I disabled kubernetes, debug mode and experimental features and the cpu usage went from 50% to ~7% - I have 3 running containers and they are idle.

**Comment by fuatkarakus** [comment]
Same issue. 
Docker version 19.03.8, build afacb8b 
Docker Desktop version 2.2.0.5 (43884)
Macos version 10.13.6

**Comment by petre-s** [comment]
Same issue, also I tried a factory reset and there is no change. ~13% CPU usage without any containers running.

**Comment by dietervdw-spotify** [comment]
How about 140% cpu usage? No containers running.

![image](https://user-images.githubusercontent.com/36199260/81328109-a376ef00-909c-11ea-9f57-74eff2f5c977.png)

**Comment by xavi-dev** [comment]
In macOS the process with hight CPU usage with no containers is hyperkit:
![image](https://user-images.githubusercontent.com/3817677/83246431-66d17b80-a1a2-11ea-82ce-c300e2fd0a43.png)

Version 2.3.0.3 (45519)

**Comment by ebriney** [comment]
Hi everybody, we are currently actively investigating on that problem but we lack some information and we need your help.
We added spindump in diagnostics in that (not yet released) build: https://desktop-stage.docker.com/mac/edge/46292/Docker.dmg
If you have the chance to got cpu locks in any of our process, can you do a diagnose and put it in: https://github.com/docker/roadmap/issues/12
Thanks.

**Comment by WimCools** [comment]
On my Mac the CPU is consumed by mDNSResolver once docker-desktop is started (even when no containers are running).
